components/sidebar/sidebar_channel/index.js

```diff
@@ -109,7 +109,7 @@ function makeMapStateToProps() {
             channelTeammateId,
             channelTeammateUsername,
             channelTeammateDeletedAt,
-            hasDraft: Boolean(draft.message || draft.fileInfos.length || draft.uploadsInProgress.length) && currentChannelId !== channel.id,
+            hasDraft: Boolean(draft.message.trim() || draft.fileInfos.length || draft.uploadsInProgress.length) && currentChannelId !== channel.id,
```

Comment:
It looks like in some cases `draft` can be a boolean

https://github.com/mattermost/mattermost-webapp/blob/6bec3f44a781948cc54976432126b102e92489e0/components/sidebar/sidebar_channel/index.js#L40

which could case this to throw a TypeError. Should probably add some defensive code for that case